pkg/engine/update.go

```diff
@@ -163,7 +163,6 @@ func HasChanges(changes display.ResourceChanges) bool {
 	for op, count := range changes {
 		if op != deploy.OpSame &&
 			op != deploy.OpRead &&
-			op != deploy.OpReadDiscard &&
 			op != deploy.OpReadReplacement {
 			c += count
```

Comment:
Similar to the display code, this changes the semantics of `HasChanges` with respect to discards of read resources. I'm not sure what exact impact this particular diff will have, but it will likely affect at least the display code.